blueflood-core/src/test/java/com/rackspacecloud/blueflood/types/BluefloodTimerRollupPercentileTest.java

```diff
@@ -0,0 +1,220 @@
+package com.rackspacecloud.blueflood.types;
+
+import org.junit.Test;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+import static org.junit.Assert.*;
+
+public class BluefloodTimerRollupPercentileTest {
+
+    @Test
+    public void maybePromoteDoubleYieldsSameValue() {
+        // given
+        Double value = 123.45d;
+        // when
+        Number result = BluefloodTimerRollup.Percentile.maybePromote(value);
+        // then
+        assertNotNull(result);
+        assertEquals(Double.class, result.getClass());
+        assertEquals(123.45d, (Double)result, 0.00001d);
+    }
+
+    @Test
+    public void maybePromoteFloatPromotesToDouble() {
+        // given
+        Float value = 123.45f;
+        // when
+        Number result = BluefloodTimerRollup.Percentile.maybePromote(value);
+        // then
+        assertNotNull(result);
+        assertEquals(Double.class, result.getClass());
+        assertEquals(123.45d, (Double)result, 0.00001d);
+    }
+
+    @Test
+    public void maybePromoteLongYieldsSameValue() {
+        // given
+        Long value = 123L;
+        // when
+        Number result = BluefloodTimerRollup.Percentile.maybePromote(value);
+        // then
+        assertNotNull(result);
+        assertEquals(Long.class, result.getClass());
+        assertEquals(123L, (long)(Long)result);
```

Comment:
@shintasmith No particular reason. In the first two tests, I had used a simple cast, so I think I was just sticking with that. I'm not entirely sure what goes on under the hood for casts or for `longValue()`, at least not in Java. Would one be better, in particular to another?